src/test/scala/is/hail/utils/HashMethodsSuite.scala

```diff
@@ -0,0 +1,20 @@
+package is.hail.utils
+
+import is.hail.SparkSuite
+import org.testng.annotations.Test
+
+class HashMethodsSuite extends SparkSuite {
+  @Test def testMultGF(): Unit = {
+    val ph = new PolyHash
+    import ph._
+
+    val x1 = 3705006673L.toInt
+    val y1 = 2551778209L.toInt
+    assert(multGF(x1, y1) == 2272553040L.toInt)
```

Comment:
i'm curious, what did you use to generate the right hand side?

Reply:
I used Mathematica to test my implementation, then just hard coded a couple of random cases.